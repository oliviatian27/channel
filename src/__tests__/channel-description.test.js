import React from 'react';
import { shallow } from 'enzyme';

import ChannelDescription from '../components/channel-list/channel-detail/channel-description';


describe('ChannelDescription tests', () => {

  it('renders ChannelDescription component', () => {
    const description = "one";
    const wrapper = shallow(<ChannelDescription description={description} />);

    // Expect the wrapper object to be defined
    expect(wrapper.find('.channel-description')).toBeDefined();
    expect(wrapper.find('.channel-description')).toMatchSnapshot();
  });

  it('displays the correct description based on props.description', () => {
    const description = "one";
    const wrapper = shallow(<ChannelDescription description={description} />);

    // Expect the description text to be props.description
    expect(wrapper.find('.channel-description').text()).toBe(description);

    // update the description props
    wrapper.setProps({description:"two"})

    expect(wrapper.find('.channel-description').text()).toBe("two");
  });
})
