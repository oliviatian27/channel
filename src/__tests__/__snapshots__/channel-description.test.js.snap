// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelDescription tests renders ChannelDescription component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <ChannelDescription
      description="one"
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          one
        </p>,
        "className": "channel-description",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "one",
        },
        "ref": null,
        "rendered": "one",
        "type": "p",
      },
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            one
          </p>,
          "className": "channel-description",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "one",
          },
          "ref": null,
          "rendered": "one",
          "type": "p",
        },
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        one
      </p>,
      "className": "channel-description",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "one",
      },
      "ref": null,
      "rendered": "one",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          one
        </p>,
        "className": "channel-description",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "one",
        },
        "ref": null,
        "rendered": "one",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          one
        </p>,
        "className": "channel-description",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "one",
        },
        "ref": null,
        "rendered": "one",
        "type": "p",
      },
      "type": "div",
    },
  ],
}
`;
